File: integrations/dart/lib/src/tokens.dart
import 'dart:js_interop';

import 'facade.dart';
import 'injector.dart';
import 'scope.dart';
import 'services.dart';
import 'token.dart';

T _facade<T extends AngularTsJsFacade>(
  JSAny? value,
  T Function(JSObject raw) create,
) {
  return create(value as JSObject);
}

/// The angular token.
final angularToken = token<Angular>(
  '\$angular',
  fromJs: (value) => _facade(value, Angular.new),
);

/// The scope token.
final scopeToken = token<Scope<Object>>(
  '\$scope',
  fromJs: Scope<Object>.unsafe,
);

/// The element token.
final elementToken = token<JSObject>('\$element');

/// The anchor scroll token.
final anchorScrollToken = token<AnchorScrollService>(
  '\$anchorScroll',
  fromJs: (value) => _facade(value, AnchorScrollService.new),
);

/// The animate token.
final animateToken = token<AnimateService>(
  '\$animate',
  fromJs: (value) => _facade(value, AnimateService.new),
);

/// The aria token.
final ariaToken = token<AriaService>(
  '\$aria',
  fromJs: (value) => _facade(value, AriaService.new),
);

/// The compile token.
final compileToken = token<CompileService>(
  '\$compile',
  fromJs: (value) => _facade(value, CompileService.new),
);

/// The cookie token.
final cookieToken = token<CookieService>(
  '\$cookie',
  fromJs: (value) => _facade(value, CookieService.new),
);

/// The controller token.
final controllerToken = token<ControllerService>(
  '\$controller',
  fromJs: (value) => _facade(value, ControllerService.new),
);

/// The document token.
final documentToken = token<JSObject>('\$document');

/// The event bus token.
final eventBusToken = token<EventBusService>(
  '\$eventBus',
  fromJs: (value) => _facade(value, EventBusService.new),
);

/// The exception handler token.
final exceptionHandlerToken = token<ExceptionHandlerService>(
  '\$exceptionHandler',
  fromJs: (value) => _facade(value, ExceptionHandlerService.new),
);

/// The filter token.
final filterToken = token<FilterService>(
  '\$filter',
  fromJs: (value) => _facade(value, FilterService.new),
);

/// The http token.
final httpToken = token<HttpService>(
  '\$http',
  fromJs: (value) => _facade(value, HttpService.new),
);

/// The http param serializer token.
final httpParamSerializerToken = token<HttpParamSerializerService>(
  '\$httpParamSerializer',
  fromJs: (value) => _facade(value, HttpParamSerializerService.new),
);

/// The interpolate token.
final interpolateToken = token<InterpolateService>(
  '\$interpolate',
  fromJs: (value) => _facade(value, InterpolateService.new),
);

/// The location token.
final locationToken = token<LocationService>(
  '\$location',
  fromJs: (value) => _facade(value, LocationService.new),
);

/// The log token.
final logToken = token<LogService>(
  '\$log',
  fromJs: (value) => _facade(value, LogService.new),
);

/// The parse token.
final parseToken = token<ParseService>(
  '\$parse',
  fromJs: (value) => _facade(value, ParseService.new),
);

/// The rest token.
final restToken = token<RestServiceFacade>(
  '\$rest',
  fromJs: (value) => _facade(value, RestServiceFacade.new),
);

/// The root scope token.
final rootScopeToken = token<Scope<Object>>(
  '\$rootScope',
  fromJs: Scope<Object>.unsafe,
);

/// The root element token.
final rootElementToken = token<JSObject>('\$rootElement');

/// The sce token.
final sceToken = token<SceService>(
  '\$sce',
  fromJs: (value) => _facade(value, SceService.new),
);

/// The sce delegate token.
final sceDelegateToken = token<SceDelegateService>(
  '\$sceDelegate',
  fromJs: (value) => _facade(value, SceDelegateService.new),
);

/// The state token.
final stateToken = token<JSObject>('\$state');

/// The state registry token.
final stateRegistryToken = token<JSObject>('\$stateRegistry');

/// The sse token.
final sseToken = token<SseService>(
  '\$sse',
  fromJs: (value) => _facade(value, SseService.new),
);

/// The template cache token.
final templateCacheToken = token<TemplateCacheService>(
  '\$templateCache',
  fromJs: (value) => _facade(value, TemplateCacheService.new),
);

/// The template factory token.
final templateFactoryToken = token<JSObject>('\$templateFactory');

/// The template request token.
final templateRequestToken = token<TemplateRequestService>(
  '\$templateRequest',
  fromJs: (value) => _facade(value, TemplateRequestService.new),
);

/// The transitions token.
final transitionsToken = token<TransitionsService>(
  '\$transitions',
  fromJs: (value) => _facade(value, TransitionsService.new),
);

/// The internal router view token.
final viewToken = token<JSObject>('\$view');

/// The window token.
final windowToken = token<JSObject>('\$window');

/// The websocket token.
final websocketToken = token<WebSocketService>(
  '\$websocket',
  fromJs: (value) => _facade(value, WebSocketService.new),
);

/// The worker token.
final workerToken = token<WorkerService>(
  '\$worker',
  fromJs: (value) => _facade(value, WorkerService.new),
);

/// The wasm token.
final wasmToken = token<WasmService>(
  '\$wasm',
  fromJs: (value) => _facade(value, WasmService.new),
);

/// The provide token.
final provideToken = token<JSObject>('\$provide');

/// The injector token.
final injectorToken = token<Injector>(
  '\$injector',
  fromJs: (value) => Injector(value as JSObject),
);

File: integrations/kotlin/src/jsMain/kotlin/angular/ts/Tokens.kt
package angular.ts

import angular.ts.generated.Angular as RawAngular
import angular.ts.generated.AnchorScrollService as RawAnchorScrollService
import angular.ts.generated.AnimateService as RawAnimateService
import angular.ts.generated.AriaService as RawAriaService
import angular.ts.generated.CompileService as RawCompileService
import angular.ts.generated.ControllerService as RawControllerService
import angular.ts.generated.CookieService as RawCookieService
import angular.ts.generated.ExceptionHandlerService as RawExceptionHandlerService
import angular.ts.generated.FilterService as RawFilterService
import angular.ts.generated.HttpParamSerializerService as RawHttpParamSerializerService
import angular.ts.generated.HttpService as RawHttpService
import angular.ts.generated.InjectorService as RawInjectorService
import angular.ts.generated.InterpolateService as RawInterpolateService
import angular.ts.generated.LocationService as RawLocationService
import angular.ts.generated.LogService as RawLogService
import angular.ts.generated.ParseService as RawParseService
import angular.ts.generated.RestFactory as RawRestFactory
import angular.ts.generated.Scope as RawScope
import angular.ts.generated.SseService as RawSseService
import angular.ts.generated.StateRegistryService as RawStateRegistryService
import angular.ts.generated.StateService as RawStateService
import angular.ts.generated.TemplateCacheService as RawTemplateCacheService
import angular.ts.generated.TemplateRequestService as RawTemplateRequestService
import angular.ts.generated.TransitionsService as RawTransitionsService
import angular.ts.generated.WebComponentService as RawWebComponentService
import angular.ts.generated.WebSocketService as RawWebSocketService
import angular.ts.generated.WebTransportService as RawWebTransportService
import org.w3c.dom.Element

public val angularToken: Token<Angular> =
    Token("\$angular") { value -> Angular(value.unsafeCast<RawAngular>()) }

public val anchorScrollToken: Token<AnchorScrollService> =
    Token("\$anchorScroll") { value -> AnchorScrollService(value.unsafeCast<RawAnchorScrollService>()) }

public val animateToken: Token<AnimateService> =
    Token("\$animate") { value -> AnimateService(value.unsafeCast<RawAnimateService>()) }

public val ariaToken: Token<AriaService> =
    Token("\$aria") { value -> AriaService(value.unsafeCast<RawAriaService>()) }

public val scopeToken: Token<Scope<Any>> =
    Token("\$scope") { value -> Scope(value.unsafeCast<RawScope>()) }

public val rootScopeToken: Token<Scope<Any>> =
    Token("\$rootScope") { value -> Scope(value.unsafeCast<RawScope>()) }

public val rootElementToken: Token<Element> =
    Token("\$rootElement") { value -> value.unsafeCast<Element>() }

public val compileToken: Token<CompileService> =
    Token("\$compile") { value -> CompileService(value.unsafeCast<RawCompileService>()) }

public val controllerToken: Token<ControllerService> =
    Token("\$controller") { value -> ControllerService(value.unsafeCast<RawControllerService>()) }

public val cookieToken: Token<CookieService> =
    Token("\$cookie") { value -> CookieService(value.unsafeCast<RawCookieService>()) }

public val exceptionHandlerToken: Token<ExceptionHandlerService> =
    Token("\$exceptionHandler") { value -> ExceptionHandlerService(value.unsafeCast<RawExceptionHandlerService>()) }

public val filterToken: Token<FilterService> =
    Token("\$filter") { value -> FilterService(value.unsafeCast<RawFilterService>()) }

public val httpToken: Token<HttpService> =
    Token("\$http") { value -> HttpService(value.unsafeCast<RawHttpService>()) }

public val httpParamSerializerToken: Token<HttpParamSerializerService> =
    Token("\$httpParamSerializer") { value ->
        HttpParamSerializerService(value.unsafeCast<RawHttpParamSerializerService>())
    }

public val injectorToken: Token<Injector> =
    Token("\$injector") { value ->
        Injector(value.unsafeCast<RawInjectorService<Any?>>())
    }

public val interpolateToken: Token<InterpolateService> =
    Token("\$interpolate") { value -> InterpolateService(value.unsafeCast<RawInterpolateService>()) }

public val logToken: Token<LogService> =
    Token("\$log") { value -> LogService(value.unsafeCast<RawLogService>()) }

public val locationToken: Token<LocationService> =
    Token("\$location") { value -> LocationService(value.unsafeCast<RawLocationService>()) }

public val parseToken: Token<ParseService> =
    Token("\$parse") { value -> ParseService(value.unsafeCast<RawParseService>()) }

public val restToken: Token<RestFactory> =
    Token("\$rest") { value -> RestFactory(value.unsafeCast<RawRestFactory>()) }

public val stateToken: Token<StateService> =
    Token("\$state") { value -> StateService(value.unsafeCast<RawStateService<Any?>>()) }

public val stateRegistryToken: Token<StateRegistryService> =
    Token("\$stateRegistry") { value -> StateRegistryService(value.unsafeCast<RawStateRegistryService>()) }

public val sseToken: Token<SseService> =
    Token("\$sse") { value -> SseService(value.unsafeCast<RawSseService>()) }

public val templateCacheToken: Token<TemplateCacheService> =
    Token("\$templateCache") { value -> TemplateCacheService(value.unsafeCast<RawTemplateCacheService>()) }

public val templateRequestToken: Token<TemplateRequestService> =
    Token("\$templateRequest") { value -> TemplateRequestService(value.unsafeCast<RawTemplateRequestService>()) }

public val transitionsToken: Token<TransitionsService> =
    Token("\$transitions") { value -> TransitionsService(value.unsafeCast<RawTransitionsService>()) }

public val websocketToken: Token<WebSocketService> =
    Token("\$websocket") { value -> WebSocketService(value.unsafeCast<RawWebSocketService>()) }

public val webTransportToken: Token<WebTransportService> =
    Token("\$webTransport") { value -> WebTransportService(value.unsafeCast<RawWebTransportService>()) }

public val webComponentToken: Token<WebComponentService> =
    Token("\$webComponent") { value -> WebComponentService(value.unsafeCast<RawWebComponentService>()) }

public val workerToken: Token<WorkerService> =
    Token("\$worker") { value -> WorkerService(value) }

public val wasmToken: Token<WasmService> =
    Token("\$wasm") { value -> WasmService(value) }
